apps/banking_api/lib/banking_api/user.ex

```diff
@@ -0,0 +1,37 @@
+defmodule BankingApi.User do
+  @moduledoc """
+  Defines the schema for users. Users have name, email, password and balance.
+  The balance must be an integer representing the value in cents.
+  When an user creates the account, he/she gets a bonus of 100000.
+  The password must have more than 6 characters.
+  THe balance must be positive or zero.
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @required_params [:name, :email, :password]
+
+  schema "users" do
+    field(:name, :string)
+    field(:email, :string)
+    field(:password, :string)
+    field(:balance, :integer, default: 100_000)
+
+    timestamps()
+  end
+
+  @doc """
+  Define validations and constraints.
+  """
+  def changeset(params, struct \\ %__MODULE__{}) do
+    struct
+    |> cast(params, [:name, :email, :password, :balance])
```

Comment:
```suggestion
    |> cast(params, __schema__(:fields))
```
Aqui podemos simplificar com essa função do `Ecto.Schema`. Vale lembrar que ela retorna uma lista de campos **não** virtuais (campos do tipo `field(:name, :string, virtual: true)` não seriam retornados), mas como aqui não temos nenhum, não precisamos nos preocupar com isso.

Docs: https://hexdocs.pm/ecto/Ecto.Schema.html#module-reflection